code/game/mob/observer/ghost/ghost.dm

```diff
@@ -163,7 +163,8 @@ var/global/list/image/ghost_sightless_images = list() //this is a list of images
 		ghost.can_reenter_corpse = can_reenter_corpse
 		ghost.timeofdeath = stat == DEAD ? timeofdeath : world.time
 		ghost.key = key
-		ghost << "<span class = 'good'><font size = 4>Or click <a href='?src=\ref[ghost];respawn=1'>THIS</a> button to respawn!</font></span>"
+		if (!(ghost.started_as_observer))
```

Comment:
Won't this also prevent admins from observing?

Reply:
It won't prevent anything. The button is just a helpfull shortcut for respawning. You can still use the command like normal